src/date-picker/date-display.jsx

```diff
@@ -41,7 +41,7 @@ function getStyles(props, state) {
       marginBottom: 10,
     },
     yearTitle: {
-      cursor: (state.selectedYear && !props.disableYearSelection) ? 'pointer' : 'default',
+      cursor: (!state.selectedYear && !props.disableYearSelection) ? 'pointer' : 'default',
```

Comment:
That doesn't appear to change the behaviour. You still get the pointer for both in Year mode, and for neither in Date mode.